### DISCUSSION TITLE
Seed data on migration

### DISCUSSION THREAD

**Discussion by mdmontesinos** [oqtane/oqtane.framework#4587] [Q&A]
Is there any way to seed table data during migrations without using raw sql queries?

I've been trying to do it but the BaseEntityBuilder does not expose the inner MigrationBuilder nor any method to insert data.

**Comment by sbwalker** [comment]
You could use the IInstallable interface. Usually its only used to execute database migrations however you are free to implement whatever logic you want:

```
        public bool Install(Tenant tenant, string version)
        {
            var migrated = Migrate(new HtmlTextContext(_DBContextDependencies), tenant, MigrationType.Up);
            if (migrated && version == "1.0.1")
            {
                // implement whatever logic you want ie. add seed data
            }
            return migrated;
        }
```

**Reply by mdmontesinos** [reply]
Right, thanks! Although I believe it would be interesting to provide support for data seeding with the entity builders, as that's where it makes the most sense. Basically providing a method that uses the internal migration builder and its InsertData method to generate InsertDataOperations that could be executed either in the Build method or in an additional Seed method.
